homeassistant/core.py

```diff
@@ -1110,6 +1110,13 @@ def __init__(
         self.domain, self.object_id = split_entity_id(self.entity_id)
         self._as_dict: ReadOnlyDict[str, Collection[Any]] | None = None
 
+    def __hash__(self) -> int:
+        """Make the state hashable.
+
+        State objects are effectively immutable.
+        """
+        return hash(id(self))
```

Comment:
Should we combine this with the `last_updated` timestamp to avoid accidental collition when Python reuses the ID. I am not sure if that can happen.

Reply:
So `id` is the memory address so reuse is entirely possible.

Good catch 👍

Reply:
aa5c41b5148ab23b9f7e9f09d0c1b22494ce9461